src/H5Pint.c

```diff
@@ -523,6 +553,28 @@ H5P__init_package(void)
     HDassert(tot_init == NELMTS(init_class));
 
 done:
+    if (ret_value < 0 && tot_init > 0) {
```

Comment:
Changes to correctly bring down the H5P interface if it failed to initialize correctly.